### PR TITLE
Update for Xcode 15.0.1/Swift 5.9/swift-syntax 509.0.2

### PR DESCRIPTION
**Pull request by shingt**
## Checklist

- [x] All tests are passed.
- [x] Added tests.
- [x] Searched existing issues to make sure not duplicated.
  - Duplicated with [this PR](https://github.com/ra1028/swift-mod/pull/22), but created this PR since it seems stale.

## Motivation and Context

Xcode 15.0.1/Swift 5.9 changes.

## Description

SwiftSyntax `509.0.x` aligns with Swift 5.9, and so adjusting to it in this PR.
Some notable changes:

* [swift-syntax 509.0.2](https://github.com/apple/swift-syntax/releases/tag/509.0.2)
  * There are a lot of deprecated interfaces, and this PR adapts to such changes.
  * It no longer requires `_InternalSwiftSyntaxParser.dylib`
* [swift-format 509.0.0](https://github.com/apple/swift-format/releases/tag/509.0.0)
  * Deleted some deprecated rules such as `BlankLineBetweenMembers`.
* First support for actor syntax. (as `classOrActorKeyword` is gone.)

For concrete changes, let me add inline comments.

### CODE REVIEW

File: Tests/SwiftModRulesTests/DefaultAccessLevelRuleTests.swift [hunk 1]
```diff
@@ -214,6 +214,18 @@ final class DefaultAccessLevelRuleTests: XCTestCase {
         )
     }
 
+    func testClassOpenNotTriggered() throws {
```
Comment (shingt) [inline Tests/SwiftModRulesTests/DefaultAccessLevelRuleTests.swift:217]: I'd like to add this test if it's ok.
(I forgot to replace `openKeyward` originally, producing the wrong output like `open open class Class` for this pattern.)

File: Makefile [hunk 2]
```diff
@@ -4,13 +4,9 @@ SWIFT_BUILD_FLAGS := -c release --disable-sandbox
...
 XCODE_DEFAULT_TOOLCHAIN := /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain
 GITHUB_REPO := ra1028/$(TOOL_NAME)
-DOCKER_IMAGE_NAME := swift:5.5
+DOCKER_IMAGE_NAME := swift:5.9
 
 ifeq ($(shell uname), Darwin)
-USE_SWIFT_STATIC_STDLIB := $(shell test -d $$(dirname $$(xcrun --find swift))/../lib/swift_static/macosx && echo use_swift_static_stdlib_flag)
-ifeq ($(USE_SWIFT_STATIC_STDLIB), use_swift_static_stdlib_flag)
-SWIFT_BUILD_FLAGS += -Xswiftc -static-stdlib
-endif
```
Comment (shingt) [inline Makefile:13]: We cannot pass `-static-stdlib` under the Darwin env anymore.
Rev: https://github.com/apple/swift-driver/pull/184

File: .swift-format.json
Comment (shingt) [inline .swift-format.json:1]: Removed deprecated rules.

File: Sources/SwiftModCore/Extensions/DeclModifierListSyntaxExtensions.swift
Comment (shingt) [inline Sources/SwiftModCore/Extensions/DeclModifierListSyntaxExtensions.swift:1]: Moved from `ModifierListSyntaxExtensions`. (As `ModifierListSyntax` is deprecated.)

File: Sources/SwiftModCore/Extensions/SyntaxFactoryExtensions.swift
Comment (shingt) [inline Sources/SwiftModCore/Extensions/SyntaxFactoryExtensions.swift:1]: Goes to `TokenSyntaxExtensions` (removing some unnecessary signatures).

File: Sources/SwiftModRules/DefaultAccessLevelRule.swift [hunk 3]
```diff
@@ -128,17 +128,17 @@ public final class DefaultAccessLevelRule: RuleDefinition {
...
+                replacingModifiers: { $0.with(\.modifiers, $1) },
                 visitChildren: nil
             )
         }
 
         public override func visit(_ node: VariableDeclSyntax) -> DeclSyntax {
             visit(
                 node,
-                getDeclKeyword: { $0.letOrVarKeyword },
+                getDeclKeyword: { $0.bindingSpecifier },
```
Comment (shingt) [inline Sources/SwiftModRules/DefaultAccessLevelRule.swift:139]: A bit unintuitive (for me), but https://github.com/apple/swift-syntax/pull/1354

File: Sources/SwiftModRules/DefaultMemberwiseInitializerRule.swift [hunk 4]
```diff
@@ -146,173 +146,162 @@ private extension DefaultMemberwiseInitializerRule.Rewriter {
...
-            leftParen: SyntaxFactory.makeLeftParenToken(),
-            parameterList: SyntaxFactory.makeFunctionParameterList(
+        let parameterClause = FunctionParameterClauseSyntax(
+            leftParen: .leftParenToken(),
+            parameters: FunctionParameterListSyntax(
                 storedProperties.indices.map { index in
                     let property = storedProperties[index]
                     let isLast = index == storedProperties.index(before: storedProperties.endIndex)
-                    return SyntaxFactory.makeFunctionParameter(
-                        attributes: nil,
```
Comment (shingt) [inline Sources/SwiftModRules/DefaultMemberwiseInitializerRule.swift:162]: Note: Removing default value arguments (in the entire this PR).

File: Sources/SwiftModCore/Extensions/IndentExtensions.swift [hunk 5]
```diff
@@ -7,7 +7,7 @@ public extension Indent {
             return .spaces(spaces)
 
         case .tab:
-            return .tabs(1)
+            return .tab
```
Comment (shingt) [inline Sources/SwiftModCore/Extensions/IndentExtensions.swift:10]: Note: Replacing with shorter trivia syntaxes like `.tab` / `.space` / etc. in this PR.

File: Sources/SwiftModRules/DefaultAccessLevelRule.swift [hunk 6]
```diff
@@ -68,17 +68,27 @@ public final class DefaultAccessLevelRule: RuleDefinition {
...
-                getDeclKeyword: { $0.classOrActorKeyword },
+                getDeclKeyword: { $0.classKeyword },
                 getModifiers: { $0.modifiers },
-                replacingModifiers: { $0.withModifiers($1) },
+                replacingModifiers: { $0.with(\.modifiers, $1) },
+                visitChildren: super.visit
+            )
+        }
+
+        public override func visit(_ node: ActorDeclSyntax) -> DeclSyntax {
```
Comment (shingt) [inline Sources/SwiftModRules/DefaultAccessLevelRule.swift:86]: Adding actor support as `classOrActorKeyword` is gone.

File: Tests/SwiftModRulesTests/DefaultAccessLevelRuleTests.swift [hunk 7]
```diff
@@ -214,6 +214,94 @@ final class DefaultAccessLevelRuleTests: XCTestCase {
...
+            source: """
+                open class Class {}
+                """,
+            expected: """
+                open class Class {}
+                """
+        )
+    }
+
+    func testActor() throws {
```
Comment (shingt) [inline Tests/SwiftModRulesTests/DefaultAccessLevelRuleTests.swift:229]: Basically same as struct tests.

File: .swift-format.json
Comment (shingt) [inline .swift-format.json:1]: Removed deprecated rules.

File: Sources/SwiftModCore/Extensions/DeclModifierListSyntaxExtensions.swift
Comment (shingt) [inline Sources/SwiftModCore/Extensions/DeclModifierListSyntaxExtensions.swift:1]: Moved from `ModifierListSyntaxExtensions`. (As `ModifierListSyntax` is deprecated.)

File: Sources/SwiftModCore/Extensions/SyntaxFactoryExtensions.swift
Comment (shingt) [inline Sources/SwiftModCore/Extensions/SyntaxFactoryExtensions.swift:1]: Goes to `TokenSyntaxExtensions` (removing some unnecessary signatures).

**Review by ra1028 — APPROVED** [review]
LGTM

**Pull request merged**